### PR TITLE
Provide p-values of selected features

### PR DESCRIPTION
**Pull request by kempa-liehr**
* FeatureSelection().relevant_features is a DataFrame which contains the p-values of the selected features. 
* FeatureSelection().relevant_features.index holds the column names of the selected features.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -1,31 +0,0 @@
-# .coveragerc to control coverage.py
```
Comment (MaxBenChrist) [inline .coveragerc]: why are you deleting the .coveragerc?

**Comment by MaxBenChrist** [comment]
this is just a change of the internal api right? what is the motivation behind the change?

**Comment by kempa-liehr** [comment]
In this particular case, I had applied the FeatureSelection transformer to my feature set, because I wanted to check the individual p-values of the selected features. This will be important for teaching purposes.

And yes, it’s just a slight change to the internal API, which simultaneously provides an external interface for retrieving the p-values of the selected features.

**Comment by kempa-liehr** [comment]
.coveragerc is generated automatically by running 

python setup.py test

And because there are no modifications to it, it’s redundant, right?

**Comment by MaxBenChrist** [comment]
On my system it does not get generated after calling `python setup.py test`

**Comment by kempa-liehr** [comment]
So, there is .coveragerc again.

**Comment by MaxBenChrist** [comment]
I think it would be a nice idea to save the df_bh DataFrame in the object, similar to `feature_importances` in for example sklearns random forest

**Comment by kempa-liehr** [comment]
Good point.

**Comment by MaxBenChrist** [comment]
You are right!

**Comment by jneuff** [comment]
This is a very good idea @kempa-liehr ! Maybe we could have an option to return the p-values or feature importances from the select_features function.

File: tsfresh/transformers/feature_selector.py [hunk 2]
```diff
@@ -82,7 +84,7 @@ def fit(self, X, y):
             y = pd.Series(y.copy())
 
         df_bh = check_fs_sig_bh(X, y, self.settings)
-        self.relevant_features = df_bh.loc[df_bh.rejected].Feature
+        self.relevant_features = df_bh.loc[df_bh.rejected].p_value
```
Comment (MaxBenChrist) [inline tsfresh/transformers/feature_selector.py]: For me, that is a public API change. (It will break some of my own scripts...)

Lets undo this changes and add the line 

`self.feature_importances_  = 1- df_bh.loc[df_bh.rejected].p_value`

Comment (MaxBenChrist) [inline tsfresh/transformers/feature_selector.py]: Also `relevant_features` sound like a name of a feature and not a numerical measurement of their importance

Comment (kempa-liehr) [inline tsfresh/transformers/feature_selector.py]: Okay, I'm going to put `self.feature_importance_` in.

Comment (kempa-liehr) [inline tsfresh/transformers/feature_selector.py]: But the change of the API does make sense, because `self.relevant_features` contains the relevant features both in its index and its values.

We could document this change for version 0.5 thoroughly and maybe even think about combining it with a diagram explaining the Benjamini-Yekutieli (cf. Fig 2 in ACML paper).

>

Comment (MaxBenChrist) [inline tsfresh/transformers/feature_selector.py]: Okay so to make it clear lets have three atrributes:

- feature_importance_ = 1-p_values, no index 
- p_values =   p_value, no index 
- relevant_features = list of relevant feature names, no index
- features = list of feature names, no index

can you change the target of this pr to the feature_extraction_settings_refactoring branch?

**Comment by kempa-liehr** [comment]
Is there a reason for dismissing the relation between feature names and p_values/significance?

File: tsfresh/transformers/feature_selector.py (hunk 2 above)
Comment (kempa-liehr) [inline tsfresh/transformers/feature_selector.py]: Is there a reason for dismissing the relation between feature names and p_values/significance?

Comment (MaxBenChrist) [inline tsfresh/transformers/feature_selector.py]: I guess it is more similar to the attributes of sklearn transformers. E.g. http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html, the feture_importance_ is just an array with the feature importances

**Comment by MaxBenChrist** [comment]
I would merge this pr. However, you are also sneaking in changes to the dritbif files. :D 

Can you undo them, so this pr will only tackle the following files:

tests/transformers/test_feature_selector.py
tsfresh/feature_extraction/feature_calculators.py
tsfresh/transformers/feature_selector.py

Thanks

**Comment by kempa-liehr** [comment]
I also skipped the changes on feature_calculators.py, because they were only minor wording corrections not related to this PR.
